Java/core/src/main/java/com/amazon/randomcutforest/tree/AbstractCompactRandomCutTree.java

```diff
@@ -87,10 +86,14 @@ public AbstractCompactRandomCutTree(int maxSize, long seed, LeafStore leafStore,
         this.rootIndex = rootIndex;
         // random = new Random(seed);
         nodeManager = new CompactNodeManager(nodeStore, leafStore, maxSize);
-        nodeView = new CompactNodeView(this, rootIndex);
         this.enableCache = enableCache;
     }
 
+    @Override
+    INode<Integer> getNodeView(Integer node) {
```

Comment:
By not including a visibility modifier (like public or protected), this method will be visible inside the tree package but not visible to any subclasses of AbstractCompactRandomCutTree. Is that intentional?

Reply:
No. These methods should be visible to sub-classes, so that other kinds of trees can be built using the same set up.